common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -109,6 +128,27 @@ public static void deleteRecursively(File file) throws IOException {
     }
   }
 
+  private static void deleteRecursivelyForUnix(File file) throws IOException {
+    Process process = null;
+    final ProcessBuilder builder = new ProcessBuilder("rm", "-rf", file.getAbsolutePath());
+
+    try {
+      process = builder.start();
+      process.waitFor();
+      if (process.exitValue() == 0 && !file.exists()) {
+        return;
+      }
+
+      throw new IOException("Failed to delete: " + file.getAbsolutePath());
```

Comment:
This is caught by the containing block. Maybe store the status and take the check above out of the try block then for clarity